src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -18,183 +31,851 @@ object HullmodExoticHandler {
+                                stats = variant.statsForOpCosts,
+                                id = exoticHullmodInstance.hullModId
+                        )
+
+                        // Now that we've uninstalled it, lets unset it from the list of installed variants and
+                        // update the lookup map
+                        // STRICT vs LENIENT - for STRICT we will reduce the 'installed on' list, for LENIENT we will not
+                        val installListToUse = if (workMode == HullmodExoticHandlerWorkMode.STRICT) {
+                            // STRICT work mode - reduce the 'installed on' list by removing the variant we will remove from
+                            val newInstalledOnList = installedOnVariants.toMutableList()
+                            newInstalledOnList.remove(variant)
+
+                            newInstalledOnList
+                        } else {
+                            // LENIENT work mode - just do nothing
+                            installedOnVariants
+                        }
+                        lookupMap[hullmodExoticKey] = installData.copy(
+                                parentFleetMemberAPI = installData.parentFleetMemberAPI,
+                                listOfExpectedVariants = installData.listOfExpectedVariants,
+                                listOfVariantsWeInstalledOn = installListToUse
+                        )
+                        logIfOverMinLogLevel("The exoticHullmod ${exoticHullmodInstance.hullModId} has been removed and lookup map has been updated. [${workMode}] Old installed list size: ${installedOnVariants.size}, new install list size: ${installListToUse.size}", Level.INFO)
+
+                        return true
+                    } else {
+                        logIfOverMinLogLevel("No ExoticHullmod with id ${hullmodId} was found !!!", Level.ERROR)
+
+                        return false
+                    }
+                } else {
+                    logIfOverMinLogLevel("The 'variant' was not in the 'installedOnVariants' list", Level.ERROR)
+
+                    return false
                 }
+            } else {
+                logIfOverMinLogLevel("There was no installed data for key: ${hullmodExoticKey}", Level.ERROR)
+
+                return false
             }
         }
-        // Log before
-        AnonymousLogger.log("removeHullmodExoticFromFleetMember()\tkeysToRemove.size: ${keysToRemove.size}\tkeysToRemove: ${keysToRemove}", "HullmodExoticHandler")
-        AnonymousLogger.log("removeHullmodExoticFromFleetMember()\tlookupMap before cleaning: ${lookupMap}", "HullmodExoticHandler")
-        AnonymousLogger.log("removeHullmodExoticFromFleetMember()\tlookupMap.size(): ${lookupMap.size}", "HullmodExoticHandler")
+    }
 
+    fun removeHullmodExoticFromFleetMember(exoticHullmodId: String, fleetMember: FleetMemberAPI) {
+        synchronized(lookupMap) {
```

Comment:
lets try without calling this method, if possible. yet, i remember it wasn't.

still, add javadoc.

Reply:
done. resolving.